test/lib/test_services.cc

```diff
@@ -156,3 +159,68 @@ test_env::impl::impl(test_env_config cfg)
 { }
 
 }
+
+static std::pair<int, char**> rebuild_arg_list_without(int argc, char** argv, const char* filter_out, bool exclude_positional_arg = false) {
+    int new_argc = 0;
+    char** new_argv = (char**) malloc(argc * sizeof(char*));
```

Comment:
You don't need to reallocate the arg list, you can just swap out unwanted items to the end, then --argc accordingly.

Reply:
That's indeed an alternative, but I found it more convenient to do any changes in a copy as the cmd separator '--' has to be restored to its original position. Also, test.py time is dominated by tests themselves, so rebuilding arg list for each test file doesn't make any real difference. Checked it verifying the elapsed time before and after. If each rebuilding takes ~5us (takes even less), then total time is 2.6k tests * ~5us = ~13ms. Not significant even if we grow number of tests to 100k, which would result in 0.5s overhead.

Reply:
With the current method, any exception will result in an ASAN leak report, possibly causing whoever has to investigate it, to start on the wrong tracks.
At the very least use `std::unique_ptr<>` to keep the allocated args alive.
The advantage of shifting arguments around is that you don't have to deal with lifetime. As long as the new arg list is shorter than the old one, you can rearrange the args any way you want.

Reply:
The arg list is automatically freed on the class destructor (see ~scylla_tests_cmdline_options_processor()).  So ASAN wise, we're good. The disadvantage with shifting arguments is that I have to undo changes, which is both tricky and error prone. The current approach is more convienient and has no significant overhead.

Reply:
I see an explicit call to `free_arg_list(new_argc, new_argv);` below. So is the arglist freed twice?

Reply:
The argument list feeded into the test is always stored in scylla_tests_cmdline_options_processor::_new_argv (see patch introducing x-log2-... option). The one destroyed explicitly is temporary and only for processing the arg list without the '--' separator. I am thinking about using defer() for automatically destroying it. Sounds good to you?

Reply:
Yes, that sounds good, although `std::unique_ptr<>` would be better, but it is indeed more clunky to use here.